src/SBInterpolatedImage.cpp

```diff
@@ -900,6 +782,198 @@ namespace galsim {
         return result;
     }
 
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    // SBInterpolatedKImage methods
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImage(
+        const BaseImage<T>& realImage,
+        const BaseImage<T>& imagImage,
+        double dk, double stepk,
+        boost::shared_ptr<Interpolant> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolated(new SBInterpolatedKImageImpl(realImage, imagImage, dk, stepk,
+                boost::shared_ptr<Interpolant2d>(new InterpolantXY(kInterp)), gsparams)
+        ) {}
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImage(
+        const BaseImage<T>& realImage,
+        const BaseImage<T>& imagImage,
+        double dk, double stepk,
+        boost::shared_ptr<Interpolant2d> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolated(new SBInterpolatedKImageImpl(realImage, imagImage, dk, stepk,
+                                                    kInterp, gsparams)
+        ) {}
+
+    SBInterpolatedKImage::SBInterpolatedKImage(const SBInterpolatedKImage& rhs)
+        : SBInterpolated(rhs) {}
+
+    SBInterpolatedKImage::~SBInterpolatedKImage() {}
+
+    ConstImageView<double> SBInterpolatedKImage::getRealKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getRealKImage();
+    }
+
+    ConstImageView<double> SBInterpolatedKImage::getImagKImage() const
+    {
+        assert(dynamic_cast<const SBInterpolatedKImageImpl*>(_pimpl.get()));
+        return static_cast<const SBInterpolatedKImageImpl&>(*_pimpl).getImagKImage();
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////
+    // SBInterpolatedKImageImpl methods
+
+    template <typename T>
+    SBInterpolatedKImage::SBInterpolatedKImageImpl::SBInterpolatedKImageImpl(
+        const BaseImage<T>& realKImage, const BaseImage<T>& imagKImage,
+        double dk, double stepk, boost::shared_ptr<Interpolant2d> kInterp,
+        const GSParamsPtr& gsparams) :
+        SBInterpolatedImpl(kInterp, stepk, 0., gsparams), //fill in maxk below
+        _dk(dk)
+    {
+        dbg<<"stepk = "<<stepk<<std::endl;
+        dbg<<"kimage bounds = "<<realKImage.getBounds()<<std::endl;
+        assert(_kInterp.get());
+
+        Ninitial = std::max(realKImage.getXMax()-realKImage.getXMin()+1,
+                            realKImage.getYMax()-realKImage.getYMin()+1);
+        dbg<<"Ninitial = "<<Ninitial<<std::endl;
+        Nk = goodFFTSize(int(Ninitial));
+        dbg<<"Nk = "<<Nk<<std::endl;
+
+        _ktab = boost::shared_ptr<KTable>(new KTable(Nk, _dk));
```

Comment:
Added the following comment (and assert) to SBInterpolatedKImageImpl constructor:
```
        // Note that _dk is the pitch of realKImage and imagKImage.  In contrast, _stepk indicates
        // the maximum pitch for drawImage() to use when rendering an image, which may be set
        // larger than _dk for efficiency.
        assert(_stepk >= _dk);
```